### ISSUE TITLE
Segment Fault - (Networking Code - PlayerbotCommandServer socket)

### ISSUE THREAD

**Issue by htc16** [ZhengPeiRu21/mod-playerbots#109]
Hello Zheng

this happens from the beginning when I installed the playerbot mod. Sometimes yes sometimes no. Command I use is .server shutdown 2m

Before saving the server and shutting down.

[gdb.txt](https://github.com/ZhengPeiRu21/mod-playerbots/files/9666469/gdb.txt)

**Comment by ZhengPeiRu21** [comment]
It looks like this is due to the PlayerbotCommandServer socket receiving a packet while it is shutting down. I am not very advanced with the networking code but I will try to solve this if I can. A good news is that this should only happen during shutdowns and should not have other negative effect.

**Comment by htc16** [comment]
yes, it's very sporadic and it only happens on the server that I have the largest database, on the smallest it doesn't.
And it's also just the times it happens.
I decided to report it because we are fixing several errors, so this is not really important in general and can be solved at any time and without haste.
